### ISSUE TITLE
More examples

### ISSUE THREAD

**Issue by mtthss** [deepmind/optax#11]
This issue provides a home for work on adding informative examples of using optax (for instance examples reproducing results from interesting optimisation papers).

Reach out on this issue if you are interested and/or have suggestions.

**Comment by PostQuantum** [comment]
I want to use optax, but I don't find any tutorials to learn. Are there any documents for me to learn optax?

**Comment by SauravMaheshkar** [comment]
Hey @mtthss can I work on this? Perhaps try to improve the existing example with RMSprop and SGD, and then move on to LRSchedulers.

**Comment by mtthss** [comment]
@SauravMaheshkar this sounds great thanks!

**Comment by SauravMaheshkar** [comment]
**Update:**
@mtthss, I've used the MNIST dataset and a `hk.module` model, and used/demonstrated how all the optimizers in the `alias.py` could be used and performed hyper-parameter optimization (using other libraries). Now I'm currently working on an example of how one could work with the second-order derivatives. Plan to make a PR by Friday 😊

**Comment by mtthss** [comment]
Awesome! Thank you very much, looking forward to this!

**Comment by mtthss** [comment]
Hi @SauravMaheshkar, how is the PR coming along? do you need any help?

**Comment by SauravMaheshkar** [comment]
Just opened a PR. 

Metrics from all the experiments are available on [this page](https://wandb.ai/sauravmaheshkar/gradient-processing-and-optimization?workspace=).

**Comment by mtthss** [comment]
Hi @SauravMaheshkar 

I am struggling a bit with interpreting the results,
do I see correctly that the train/test accuracy is around 10%?

**Comment by SauravMaheshkar** [comment]
Yes @mtthss, I can't figure that out as well. I've ported the code for training from a [Haiku Example](https://github.com/deepmind/dm-haiku/blob/master/examples/mnist.py) and updated it to use `optax` syntax. 

The only reason that I can think of is perhaps I'm training for fewer epochs. The original example uses `10,000` epochs whereas I've used `{50, 100, 200}`.

What do you think ?

**Comment by SauravMaheshkar** [comment]
Just Updating on my Progress. I've been running experiments with 10000 updates. Will open a PR once again, in the same format. Anything I should keep in mind?

**Comment by mtthss** [comment]
Awesome! Will take a look as soon as you send the new PR!

**Comment by charmerDark** [comment]
Hey ! I am interested in trying to reproduce competitive gradient Descent from [Schafer et al](https://arxiv.org/abs/1905.12103). Could I give it a try?

**Comment by mtthss** [comment]
@charmerDark hey, are you still interested in implementing this?
We are closing this "catch all" issue, but feel free to open an issue specifically for making an example reproducing competitive gradient Descent?
